packages/spright-components/src/chat/message/index.ts

```diff
@@ -22,7 +35,23 @@ export class ChatMessage extends FoundationElement {
      */
     @attr({ attribute: 'message-type' })
     public readonly messageType: ChatMessageType = ChatMessageType.system;
+
+    /** @internal */
+    @observable
+    public footerActionsIsEmpty = true;
+
+    /** @internal */
+    @observable
+    public readonly slottedFooterActionsElements?: HTMLElement[];
+
+    public slottedFooterActionsElementsChanged(
+        _prev: HTMLElement[] | undefined,
+        next: HTMLElement[] | undefined
+    ): void {
+        this.footerActionsIsEmpty = !next?.length;
```

Comment:
This pattern (and the corresponding code in the template which updates the classes on the element based on this attribute) is similar to what the FAST `endSlotTemplate` does (it removes the `end` class from the element when nothing is slotted). We also do this in the [Nimble dialog footer](https://github.com/ni/nimble/blob/c37746a42e97c24a46588f7fd3ae6f9f730a2a2e/packages/nimble-components/src/dialog/index.ts#L60C1-L67C1).

Reply:
When it has baseline support we should consider using `:has-slotted` instead. Filed this tech debt to track it: #2579